src/App.js

```diff
@@ -1,23 +1,90 @@
-import logo from './logo.svg';
-import './App.css';
-
+import React , {useEffect} from 'react'
+import ProductList from './components/ProductList'
+import WishList from './components/WishList'
+import Home from './components/Home'
+import Cart from './components/Cart'
+import Productpage from './components/Productpage'
+import Login from './components/Login'
+import {Routes , Route}from 'react-router-dom'
+import './styles/App.css'
+import Filter from './components/Filter'
+import Sort from './components/Sort'
+import SignUp from './components/SignUp'
+import {PrivateRoute} from './util'
+import axios from 'axios'
+import { useData } from './contexts/DataContext'
+import { useAuth } from './contexts/AuthContext'
 function App() {
+  const {login , setUsers , userId} = useAuth()
+  const {dispatch} = useData()
+  //const id = JSON.parse(localStorage?.getItem('userId'))?.id
+  useEffect(()=>{
+    (async () => {
+        try {
+            dispatch({type : 'LOADING_STATUS' , payload : true});
+            const {data : {products}} = await axios.get('https://intense-scrubland-09454.herokuapp.com/products')
+            dispatch({type : 'DATA' , payload : products})
+        } catch (error) {
+            console.log(error)
+        }
+        finally {
+            dispatch({type : 'LOADING_STATUS' , payload : false});
+        }
+    })(); 
+},[dispatch])
```

Comment:
Why ,[dispatch] ? , 
load the products only ones!

Reply:
Giving an warning add dispatch or remove dependency array